rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp

```diff
@@ -188,6 +188,12 @@ class SubscriberWithTopicStatistics : public rclcpp::Node
     auto options = rclcpp::SubscriptionOptions();
     options.topic_stats_options.state = rclcpp::TopicStatisticsState::Enable;
     options.topic_stats_options.publish_period = publish_period;
+    // There is a possibility that age and period messages are published almost at the
+    // same time, and if the message order is inverted when storing DDS HistoryCache
+    // 1st message could be lost with system default QoS keep_last.
+    // We explicitly set the QoS with keep_all for the test, so that both messages
+    // will be stored in the HistoryCache to be sent out.
```

Comment:
the regression can be observed only with rmw_connextdds, but this setting does not do any harm to any other rmw implementation for the test.